src/streams.c

```diff
@@ -81,67 +83,121 @@ static Int READ_COMMAND(Obj *evalResult)
 
 
 /*
- * This function reads all code from the stream and returns either `fail` if the stream
- * cannot be opened, or executes all statements in the stream and returns a list of lists,
- * each entry of which reflects the result of the execution of one statement.
+ * This function reads all code from the stream and returns either `fail` if
+ * the stream cannot be opened, or executes all statements in the stream and
+ * returns a list of lists, each entry of which reflects the result of the
+ * execution of one statement.
  *
- * The results are returned as lists with at most three bound positions.
+ * The results are returned as lists of length three if outputFunc is
+ * false, or five, if outputFunc is a function.
  *
- * If the first entry is `true`, then the second entry is bound to the result of
- * the statement if there was one, and unbound otherwise, and the third entry
- * reflects whether the statement ended in a dual semicolon,
- * if the first entry is `false', an error occurred during the execution of a statement,
- * and no other positions of the list are bound.
+ * If the first entry is `true`, then the second entry is bound to the result
+ * of the statement if there was one, and unbound otherwise, and the third
+ * entry reflects whether the statement ended in a dual semicolon, if the
+ * first entry is `false', an error occurred during the execution of a
```

Comment:
Elsewhere you use MarkDown syntax, like
```
`false`
```
but here TeX-style"
```
`false'
```

But let's not worry too much about it: This documentation comment completely violates the "standard" we use everywhere else in the kernel, so I'd be tempted to rewrite it in a follow up commit anyway. The content of this comment is much more important!